File: apps/plant_shop_c-sharp/plant_shop_asp_EF_core/Utils/PasswordUtil.cs
namespace plant_shop_asp_EF_core.Utils
{
    /// <summary>
    /// Utilitaire de hachage de mots de passe.
    /// </summary>
    public static class PasswordUtil
    {
        /// <summary>
        /// Hache un mot de passe avec BCrypt.
        /// </summary>
        /// <param name="password">Mot de passe en clair</param>
        /// <returns>Hash BCrypt</returns>
        public static string HashPassword(string password)
        {
            return BCrypt.Net.BCrypt.HashPassword(password);
        }

        /// <summary>
        /// Verifie un mot de passe contre un hash.
        /// </summary>
        /// <param name="password">Mot de passe en clair</param>
        /// <param name="hashedPassword">Hash BCrypt</param>
        /// <returns>true si correspondance</returns>
        public static bool CheckPassword(string password, string hashedPassword)
        {
            try
            {
                return BCrypt.Net.BCrypt.Verify(password, hashedPassword);
            }
            catch
            {
                return false;
            }
        }
    }
}

File: apps/plant_shop_java/Structure_en_micro_services/plant_shop_java_spring_microservices/plant_shop_java_spring_boot_security_microservices/utils/src/PasswordUtil.java
package util;

import org.mindrot.jbcrypt.BCrypt;

/** Utilitaire de hachage de mots de passe avec BCrypt */
public final class PasswordUtil {

    /** Constructeur prive (classe utilitaire) */
    private PasswordUtil() {}

    /** Hache un mot de passe en clair */
    public static String hashPassword(String plainTextPassword) {
        return BCrypt.hashpw(plainTextPassword, BCrypt.gensalt());
    }

    /** Verifie un mot de passe contre son hash */
    public static boolean checkPassword(String plainTextPassword, String hashedPassword) {
        if (plainTextPassword == null || hashedPassword == null) {
            return false;
        }
        try {
            return BCrypt.checkpw(plainTextPassword, hashedPassword);
        } catch (IllegalArgumentException e) {
            System.err.println("Avertissement : hash BCrypt invalide.");
            return false;
        }
    }
}
